x-pack/plugins/ml/public/application/jobs/components/custom_url_editor/utils.js

```diff
@@ -289,64 +286,53 @@ export function getTestUrl(job, customUrl) {
-          testUrl = replaceTokensInUrlValue(customUrl, bucketSpanSecs, record, 'timestamp');
-          resolve(testUrl);
-        } else {
-          // No anomalies yet for this job, so do a preview of the search
-          // configured in the job datafeed to obtain sample docs.
-          mlJobService.searchPreview(job).then((response) => {
-            let testDoc;
-            const docTimeFieldName = job.data_description.time_field;
-
-            // Handle datafeeds which use aggregations or documents.
-            if (response.aggregations) {
-              // Create a dummy object which contains the fields necessary to build the URL.
-              const firstBucket = response.aggregations.buckets.buckets[0];
-              testDoc = {
-                [docTimeFieldName]: firstBucket.key,
-              };
-
-              // Look for bucket aggregations which match the tokens in the URL.
-              urlValue.replace(/\$([^?&$\'"]{1,40})\$/g, (match, name) => {
-                if (name !== 'earliest' && name !== 'latest' && firstBucket[name] !== undefined) {
-                  const tokenBuckets = firstBucket[name];
-                  if (tokenBuckets.buckets) {
-                    testDoc[name] = tokenBuckets.buckets[0].key;
-                  }
-                }
-              });
-            } else {
-              if (response.hits.total.value > 0) {
-                testDoc = getProcessedFields(response.hits.hits[0].fields);
-              }
-            }
-
-            if (testDoc !== undefined) {
-              testUrl = replaceTokensInUrlValue(
-                customUrl,
-                bucketSpanSecs,
-                testDoc,
-                docTimeFieldName
-              );
-            }
+  let resp;
+  try {
+    resp = await ml.results.anomalySearch(
+      {
+        body,
+      },
+      [job.job_id]
+    );
+  } catch (error) {
+    // search may fail if the job doesn't already exist
+  }
 
-            resolve(testUrl);
-          });
-        }
-      })
-      .catch((resp) => {
-        reject(resp);
-      });
-  });
+  if (resp && resp.hits.total.value > 0) {
+    const record = resp.hits.hits[0]._source;
+    testUrl = replaceTokensInUrlValue(customUrl, bucketSpanSecs, record, 'timestamp');
+    return testUrl;
+  } else {
+    // No anomalies yet for this job, so do a preview of the search
+    // configured in the job datafeed to obtain sample docs.
+
+    let { datafeed_config: datafeedConfig, ...jobConfig } = job;
+    try {
+      // attempt load the non-combined job and datafeed so they can be used in the datafeed preview
+      const [{ jobs }, { datafeeds }] = await Promise.all([
+        ml.getJobs({ jobId: job.job_id }),
+        ml.getDatafeeds({ datafeedId: job.datafeed_config.datafeed_id }),
+      ]);
+      datafeedConfig = datafeeds[0];
+      jobConfig = jobs[0];
+    } catch (error) {
+      // jobs may not exist as this might be called from the AD job wizards
+    }
```

Comment:
do we want to check for a particular error code/message here? otherwise, it might silently fail unintentionally.